From 8fc435a94ab855459979b97f800a2ff542d15ca6 Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Mon, 14 Dec 2015 15:38:47 +0800
Subject: [PATCH 13/13] atomisp2:Add conditional compilation for YUV420_16 &
 YCgCo444_16

the patch will add to support conditional compilation for
YUV420_16 & YCgCo444_16.Current,the YUV420_16 & YCgCo444_16 only
be used ISP2.7 in BTNS platform.

Change-Id: I2270182cb69a6f1f3a34e8427642634ceda8b2ec
Original-Tracked-On: #H1604058903
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-13009
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/89810
Reviewed-by: DaleiX Liu <daleix.liu@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Build: Dessert M COHO (SC)
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Reviewed-on: https://android.intel.com:443/450394
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c    |  8 ++++++++
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c  | 10 ++++++++--
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c |  4 ++++
 3 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 086e918..6338722 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1931,10 +1931,14 @@ v4l2_fmt_to_sh_fmt(u32 fmt)
 	case V4L2_PIX_FMT_JPEG:
 	case V4L2_PIX_FMT_CUSTOM_M10MO_RAW:
 		return CSS_FRAME_FORMAT_BINARY_8;
+#ifdef V4L2_PIX_FMT_CUSTOM_YUV420_16
 	case V4L2_PIX_FMT_CUSTOM_YUV420_16:
 		return CSS_FRAME_FORMAT_YUV420_16;
+#endif
+#ifdef V4L2_PIX_FMT_CUSTOM_YCgCo444_16
 	case V4L2_PIX_FMT_CUSTOM_YCgCo444_16:
 		return CSS_FRAME_FORMAT_YCgCo444_16;
+#endif
 	case V4L2_PIX_FMT_SBGGR16:
 	case V4L2_PIX_FMT_SBGGR10:
 	case V4L2_PIX_FMT_SGBRG10:
@@ -2008,8 +2012,12 @@ static u32 get_pixel_depth(u32 pixelformat)
 	case V4L2_PIX_FMT_SGBRG10:
 	case V4L2_PIX_FMT_SGRBG10:
 	case V4L2_PIX_FMT_SRGGB10:
+#ifdef V4L2_PIX_FMT_CUSTOM_YUV420_16
 	case V4L2_PIX_FMT_CUSTOM_YUV420_16:
+#endif
+#ifdef V4L2_PIX_FMT_CUSTOM_YCgCo444_16
 	case V4L2_PIX_FMT_CUSTOM_YCgCo444_16:
+#endif
 		return 16;
 	case V4L2_PIX_FMT_RGB24:
 	case V4L2_PIX_FMT_YUV444:
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 89ad3b9..4e692e9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -516,20 +516,26 @@ const struct atomisp_format_bridge atomisp_output_fmts[] = {
 		.mbus_code = V4L2_MBUS_FMT_CUSTOM_M10MO_RAW,
 		.sh_fmt = CSS_FRAME_FORMAT_BINARY_8,
 		.description = "Custom RAW for M10MO"
-	}, {
+	},
+#ifdef V4L2_PIX_FMT_CUSTOM_YCgCo444_16
+	{
 	/* Only for ISP2.7 format*/
 		.pixelformat = V4L2_PIX_FMT_CUSTOM_YCgCo444_16,
 		.depth = 16,
 		.mbus_code = V4L2_MBUS_FMT_CUSTOM_YCgCo444_16,
 		.sh_fmt = CSS_FRAME_FORMAT_YCgCo444_16,
 		.description = "Custom YCgCo444_16"
-	}, {
+	},
+#endif
+#ifdef V4L2_PIX_FMT_CUSTOM_YUV420_16
+	{
 		.pixelformat = V4L2_PIX_FMT_CUSTOM_YUV420_16,
 		.depth = 16,
 		.mbus_code = V4L2_MBUS_FMT_CUSTOM_YUV420_16,
 		.sh_fmt = CSS_FRAME_FORMAT_YUV420_16,
 		.description = "Custom YUV420_16"
 	},
+#endif
 };
 
 const struct atomisp_format_bridge *atomisp_get_format_bridge(
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
index af58232..e4d4a80 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
@@ -57,8 +57,12 @@ const struct atomisp_in_fmt_conv atomisp_in_fmt_conv[] = {
 	/* no valid V4L2 MBUS code for metadata format, so leave it 0. */
 	{ 0, 0, 0, ATOMISP_INPUT_FORMAT_EMBEDDED, 0, IA_CSS_STREAM_FORMAT_EMBEDDED },
 	/* only for ISP2.7*/
+#ifdef V4L2_MBUS_FMT_CUSTOM_YCgCo444_16
 	{ V4L2_MBUS_FMT_CUSTOM_YCgCo444_16, 16, 16, CSS_FRAME_FORMAT_YCgCo444_16, 0, CSS_FRAME_FORMAT_YCgCo444_16 },
+#endif
+#ifdef V4L2_MBUS_FMT_CUSTOM_YUV420_16
 	{ V4L2_MBUS_FMT_CUSTOM_YUV420_16, 16, 16, CSS_FRAME_FORMAT_YUV420_16, 0, CSS_FRAME_FORMAT_YUV420_16 },
+#endif
 	{}
 };
 
-- 
1.9.1

